Priority Queue/SimplePriorityQueue.cs

```diff
@@ -462,6 +491,50 @@ public bool TryRemove(TItem item)
             }
         }
 
+        /// <summary>
+        /// Attempts to remove an item from the queue. The item does not need to be the head of the queue.
+        /// Useful for multi-threading, where the queue may become empty between calls to Contains(), GetPriority() and Remove()
+        /// Returns true if the item was successfully removed, false if it wasn't in the queue.
+        /// Assigns priority to priority of removed item if successfully removed.
+        /// If multiple copies of the item are enqueued, only the first one is removed. 
+        /// O(log n)
+        /// </summary>
+        public bool TryRemove(TItem item, out TPriority priority)
```

Comment:
For clarity/consistency the `priority` parameter here should probably be named `itemPriority`, but leaving this code review as is for now.